### 1. Из колоды в 52 карты извлекаются случайным образом 4 карты. 

#### a) Найти вероятность того, что все карты – крести: 

In [3]:
from math import factorial

def сombination (n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

Находим сначало общее число вариантов раздачи 4-х карт из 52-ти, по следующей формуле:

$$C_n^k = \frac{n!}{k!(n - k)!}$$

В нашем случае получается следующее:

$$C_4^{52} = \frac{52!}{4!(52 - 4)!} = 270725$$

В python-коде это будет выглядеть так:

In [5]:
n = сombination(52, 4)
print(f'Общее число исходов: {n}')

Общее число исходов: 270725


Теперь находим общее число вариантов, когда все 4 карты при раздаче оказались крести:

$$C_4^{13} = \frac{13!}{4!(13 - 4)!} = 715$$

In [6]:
# В коде это будет выглядеть так:

m = сombination(13, 4)
print(f'Общее число исходов, когда все из 4-х карт крести: {m}')

Общее число исходов, когда все из 4-х карт крести: 715


И наконец находим вероятность выпадения единственного варианта, когда на раздаче все 4 карты оказались крести:

$$P = \frac{m}{n} = \frac{715}{270725} = 0.00264$$

In [7]:
p = m/n
print(f'Вероятность вытащить из колоды 4 карты масти крести: {p}')

Вероятность вытащить из колоды 4 карты масти крести: 0.0026410564225690276


#### Ответ: вероятность получить при раздаче 4-х крестей из колоды в 52-х карт: 0.00264 или 0.264%.

#### б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз:

In [25]:
# 1. Находим подмножество без тузов: 52 карты - 4 туза = 48 карт
# 2. Находим число сочетаний из 4-х карт в подмножестве из 48 карт:

c = сombination(48, 4)
print(f'Общее число исходов с, когда из 48 карт нет ни одного туза: {с}')

Общее число исходов с, когда из 48 карт нет ни одного туза: 194580


In [26]:
# 3. Находим число сочетаний, когда из 4-х карт есть хотя бы один туз: 

d = n - с
print(f'Общее число исходов "d", когда в 4-х картах из 52-х хотя бы один - туз: {d}')

Общее число исходов "d", когда в 4-х картах из 52-х хотя бы один - туз: 76145


In [28]:
# 4) Находим вероятность того, что среди 4-х карт окажется хотя бы один туз (P(A)):
p = d / n
print(f'Вероятность вытащить четыре карты хотя бы с одним тузом: {p}')

Вероятность вытащить четыре карты хотя бы с одним тузом: 0.2812632745405855


#### Ответ: вероятность получить из колоды в 4 карты хотя бы одного туза: 0.28126 или 28.13%.

### 2. На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?

In [11]:
n = сombination(10, 3)
print(f'Общее количество сочетаний из 3-х цифр от 0 до 9: {n}')

Общее количество сочетаний из 3-х цифр от 0 до 9: 120


In [12]:
p =1/n
print(f'Вероятность угадать код с первой попытки: {p}')

Вероятность угадать код с первой попытки: 0.008333333333333333


#### Ответ: вероятность случайно угадать код с первой попытки: 0.00833 или 0.833%

### 3. В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?

In [13]:
n = сombination(15, 3)
print(f'Общее число исходов: {n}')

Общее число исходов: 455


In [14]:
m = сombination(9, 3)
print(f'Количество сочетаний из 3-х окрашенных деталей: {m}')

Количество сочетаний из 3-х окрашенных деталей: 84


In [16]:
p = m/n
print(f'Вероятность вытащить 3 окрашенные детали: {p}')

Вероятность вытащить 3 окрашенные детали: 0.18461538461538463


#### Ответ: вероятность вытащить 3 окрашенные детали: 0.18462 или 18.461%

### 4. В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?

In [17]:
n = сombination(100, 2)
print(f'Общее число комбинаций, из 2-х билетов: {n}')

Общее число комбинаций, из 2-х билетов: 4950


In [18]:
m = сombination(2, 2)
print(f'Количество выигрышных вариантов: {m}')

Количество выигрышных вариантов: 1


In [20]:
p = m/n
print(f'Вероятность вытащить 2 выигрышных билета: {p}')

Вероятность вытащить 2 выигрышных билета: 0.00020202020202020202


#### Ответ: вероятность купить сразу 2 (!) выигрышных билета с одной попытки (!): 0.00020 или 0.020%